Mathlib/Data/Finsupp/Basic.lean

```diff
@@ -722,6 +727,29 @@ section Zero
 
 variable [Zero M]
 
+lemma embDomain_comapDomain {f : α ↪ β} {g : β →₀ M} (hg: ↑g.support ⊆ Set.range ⇑f) :
+    embDomain f (comapDomain (⇑f) g (f.injective.injOn _)) = g := by
+  ext b
+  by_cases hb : b ∈ Set.range f
+  · rcases Set.mem_range.1 hb with ⟨a, rfl⟩
+    rw [embDomain_apply, comapDomain_apply]
+  · rw [embDomain_notin_range _ _ _ hb]
+    apply symm
+    rw [← not_mem_support_iff]
+    by_contra hb'
+    exact hb (hg hb')
+
+lemma exists_mem_embDomain_eq {f : α ↪ β} {g : β →₀ M} :
+    (∃ (v : α →₀ M), embDomain f v = g) ↔ (g.support : Set β) ⊆ (Set.range f) := by
+  constructor
+  · rintro ⟨v, rfl⟩
+    simp only [Finsupp.support_embDomain, coe_map, Set.image_subset_iff,
+      Set.preimage_range, Set.subset_univ]
+  · intro hg
+    use Finsupp.comapDomain f g (f.injective.injOn _)
+    ext b
+    rw [embDomain_comapDomain hg]
```

Comment:
I'm not convinced this lemma is worth having; it's not longer used, and the statement feels rather contrived. Do you agree?

Reply:
I see that it isn't used anymore, but we have the extension by 0 functor, and I tend to like that we can describe its image, which is given by the <- part
(the -> direction  is just `simp`).
It is true that anybody fluent enough in mathlib would think of writing `Finsupp.comapDomain f g (f.injective.injOn _)` when in need of that map?
If you think yes, then OK. (And precisely, the functions you wrote may make that true.)

Reply:
I would think yes. `comapDomain` is rather descriptive as a name.